test_pull_requests

```diff
@@ -461,31 +413,107 @@ class FlushableStore
   end
 end
 
-# Quick-and-dirty middleware that attempts to invalidate appropriate
-# cache entries before handing the request off to the
-# Faraday::HttpCache middleware
-class GitHubCacheInvalidator < Faraday::Middleware
 
-  UNSAFE_METHODS = [:post, :put, :delete, :patch].freeze
+class InvalidatingGitHubCache < Faraday::HttpCache
+
+  class CustomInvalidatingStorage < Faraday::HttpCache::Storage
+
+    attr_accessor :invalidatable
+
+    def initialize(*args)
+      super
+      @invalidatable = Hash.new { |h,k| h[k] = {} }
+      @cache.storage = self
+    end
+
+    # GitHub-specific (although probably other APIs are similar) method
+    # to remove the pagination parameters from a URL so it can be better
+    # classified
+    #
+    # Example:
+    #   synthesize_url("https://api.github.com/some/resource/id?per_page=100")
+    #     returns
+    #   "https://api.github.com/some/resource/id"
+    def synthesize_url(orig_url)
```

Comment:
Hmm, not sure -- do you have examples of URLs that would make use of both code paths? As is it seems like you would either strip out query params OR genericize